frontend/app/src/store/balances/index.ts

```diff
@@ -90,18 +90,23 @@ export const useBalancesStore = defineStore('balances', () => {
   };
 
   const fetch = async (): Promise<void> => {
-    await fetchManualBalances();
-    await priceStore.fetchExchangeRates();
-    await fetchBalances();
-    await refreshAccounts();
-    await fetchConnectedExchangeBalances();
-    await fetchNonFungibleBalances();
+    await Promise.allSettled([
+      fetchManualBalances(),
+      fetchExchangeRates(),
+      fetchBalances(),
+      refreshAccounts(),
+      fetchConnectedExchangeBalances(),
+      fetchNonFungibleBalances()
+    ]);
```

Comment:
```suggestion
    await fetchExchangeRates();
    await fetchBalances();
    await Promise.allSettled([
      fetchManualBalances(),
      refreshAccounts(),
      fetchConnectedExchangeBalances(),
      fetchNonFungibleBalances()
    ]);
```

It would be better to keep exchange rates sync first and then `fetchBalances` is the request that takes the snapshot so it should run before the others.